internal/runbits/requirements/requirements.go

```diff
@@ -441,56 +538,88 @@ func (r *RequirementOperation) shouldPromptForSecurity(vulnerabilities *model.Vu
 	logging.Debug("Prompt level: ", promptLevel)
 	switch promptLevel {
 	case vulnModel.SeverityCritical:
-		return len(vulnerabilities.Critical) > 0
+		return vulnerabilities.Critical.Count > 0
 	case vulnModel.SeverityHigh:
-		return (len(vulnerabilities.Critical) > 0 ||
-			len(vulnerabilities.High) > 0)
+		return vulnerabilities.Critical.Count > 0 ||
+			vulnerabilities.High.Count > 0
 	case vulnModel.SeverityMedium:
-		return (len(vulnerabilities.Critical) > 0 ||
-			len(vulnerabilities.High) > 0 ||
-			len(vulnerabilities.Medium) > 0)
+		return vulnerabilities.Critical.Count > 0 ||
+			vulnerabilities.High.Count > 0 ||
+			vulnerabilities.Medium.Count > 0
 	case vulnModel.SeverityLow:
-		return (len(vulnerabilities.Critical) > 0 ||
-			len(vulnerabilities.High) > 0 ||
-			len(vulnerabilities.Medium) > 0 ||
-			len(vulnerabilities.Low) > 0)
+		return vulnerabilities.Critical.Count > 0 ||
+			vulnerabilities.High.Count > 0 ||
+			vulnerabilities.Medium.Count > 0 ||
+			vulnerabilities.Low.Count > 0
 	}
 
 	return false
 }
 
-func (r *RequirementOperation) promptForSecurity(out output.Outputer, vulnerabilities *model.Vulnerabilites) (bool, error) {
-	out.Notice(locale.Tr("warning_vulnerable_simple", strconv.Itoa(vulnerabilities.Length())))
+func (r *RequirementOperation) summarizeCVEs(out output.Outputer, vulnerabilities model.VulnerableIngredientsByLevels) {
+	out.Print("")
 
-	var pkgVersionVulns []string
-	if len(vulnerabilities.Critical) > 0 {
-		criticalOutput := fmt.Sprintf("[RED]%d Critical: [/RESET]", len(vulnerabilities.Critical))
-		criticalOutput += fmt.Sprintf("[CYAN]%s[/RESET]", strings.Join(vulnerabilities.Critical, ", "))
-		pkgVersionVulns = append(pkgVersionVulns, criticalOutput)
+	if vulnerabilities.CountPrimary == 0 {
+		out.Print(locale.Tr("warning_vulnerable_indirectonly", strconv.Itoa(vulnerabilities.Count)))
+	} else if vulnerabilities.CountPrimary == vulnerabilities.Count {
+		out.Print(locale.Tr("warning_vulnerable_directonly", strconv.Itoa(vulnerabilities.Count)))
+	} else {
+		out.Print(locale.Tr("warning_vulnerable", strconv.Itoa(vulnerabilities.CountPrimary), strconv.Itoa(vulnerabilities.Count-vulnerabilities.CountPrimary)))
 	}
 
-	if len(vulnerabilities.High) > 0 {
-		highOutput := fmt.Sprintf("[ORANGE]%d High: [/RESET]", len(vulnerabilities.High))
-		highOutput += fmt.Sprintf("[CYAN]%s[/RESET]", strings.Join(vulnerabilities.High, ", "))
-		pkgVersionVulns = append(pkgVersionVulns, highOutput)
+	if vulnerabilities.Critical.Count > 0 {
```

Comment:
nit: a lot of repeated logic with each vulnerability level that could be helper functions.